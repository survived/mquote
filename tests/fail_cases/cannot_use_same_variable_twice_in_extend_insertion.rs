extern crate mquote;

use mquote::mquote;

fn main() {
    let insertion = mquote!(h e l l o);
    let q = mquote!(let _ = [ ^{insertion} ^{insertion} ]);
    assert_eq!(q.to_string(), "let _ = [ h e l l o ]");
}
