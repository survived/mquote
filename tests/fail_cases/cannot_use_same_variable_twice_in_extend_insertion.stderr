error[E0382]: use of moved value: `insertion`
 --> $DIR/cannot_use_same_variable_twice_in_extend_insertion.rs:7:46
  |
6 |     let insertion = mquote!(h e l l o);
  |         --------- move occurs because `insertion` has type `mquote::__rt::proc_macro2::TokenStream`, which does not implement the `Copy` trait
7 |     let q = mquote!(let _ = [ ^{insertion} ^{insertion} ]);
  |                                 ---------    ^^^^^^^^^ value used here after move
  |                                 |
  |                                 value moved here

For more information about this error, try `rustc --explain E0382`.
